src/bp-media/bp-media-settings.php

```diff
@@ -1923,64 +1923,87 @@ function bp_media_settings_callback_extension_video_support() {
  */
 function bb_media_settings_callback_symlink_support() {
 
+	$serve_from_server = false;
+	$offload_media     = false;
+	if ( class_exists( 'Amazon_S3_And_CloudFront' ) ) {
+		$wp_offload_media  = bp_get_option( Amazon_S3_And_CloudFront::SETTINGS_KEY );
+		$offload_media     = ( isset( $wp_offload_media ) && isset( $wp_offload_media['copy-to-s3'] ) && (bool) $wp_offload_media['copy-to-s3'] );
+		$delivery_provider = ! empty( $wp_offload_media['delivery-provider'] ) && 'storage' !== $wp_offload_media['delivery-provider'] ? ucwords( $wp_offload_media['delivery-provider'] ) : 'Amazon S3';
+		$serve_from_server = ( isset( $wp_offload_media ) && isset( $wp_offload_media['serve-from-s3'] ) && (bool) $wp_offload_media['serve-from-s3'] );
+	}
```

Comment:
@yudhisthirnahar 
Please move this changes to bp-wp-offload-media-helpers.php this file. use action for the same.